hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/storage/AsyncInventoryStorage.java

```diff
@@ -0,0 +1,444 @@
+                Relationship.Blueprint bp = new Relationship.Blueprint(Direction.outgoing, incorporates.toString(),
+                        metricPath, metric.getProperties());
+                relationship(resourceCanonicalPath, bp);
+            }
+            Collection<? extends AvailInstance<?, ?, ?>> availInstances = resource.getAvails();
+            for (AvailInstance<?, ?, ?> metric : availInstances) {
+                metric(metric);
+                CanonicalPath metricPath = newPathPrefix().metric(getInventoryId(metric)).get();
+                Relationship.Blueprint bp = new Relationship.Blueprint(Direction.outgoing, incorporates.toString(),
+                        metricPath, metric.getProperties());
+                relationship(resourceCanonicalPath, bp);
+            }
+
+        }
+
+        public BulkPayloadBuilder resourceType(ResourceType<?, ?, ?, ?> resourceType) {
+
+            org.hawkular.inventory.api.model.ResourceType.Blueprint blueprint = //
+            new org.hawkular.inventory.api.model.ResourceType.Blueprint(getInventoryId(resourceType),
+                    resourceType.getProperties());
+            entity(blueprint, org.hawkular.inventory.api.model.ResourceType.class);
+
+            CanonicalPath parentPath = newPathPrefix().resourceType(getInventoryId(resourceType)).get();
+
+            Collection<? extends MetricType> metricTypes = resourceType.getMetricTypes();
+            for (MetricType metricType : metricTypes) {
+                metricType(metricType);
+
+                String metricTypeId = getInventoryId(metricType);
+
+                CanonicalPath metricTypePath = newPathPrefix().metricType(metricTypeId).get();
+                Relationship.Blueprint bp = new Relationship.Blueprint(Direction.outgoing, incorporates.toString(),
+                        metricTypePath, Collections.emptyMap());
+                relationship(parentPath, bp);
+            }
+
+            Collection<? extends AvailType> availTypes = resourceType.getAvailTypes();
+            for (AvailType availType : availTypes) {
+                metricType(availType);
+
+                String metricTypeId = getInventoryId(availType);
+                CanonicalPath metricTypePath = newPathPrefix().metricType(metricTypeId).get();
+                Relationship.Blueprint bp = new Relationship.Blueprint(Direction.outgoing, incorporates.toString(),
+                        metricTypePath, Collections.emptyMap());
+                relationship(parentPath, bp);
+            }
+
+            return this;
+        }
+
+        private String toKey(Class<? extends AbstractElement<?, ?>> cl) {
+            String src = cl.getSimpleName();
+            return new StringBuilder(src.length()).append(Character.toLowerCase(src.charAt(0))).append(src.substring(1))
+                    .toString();
+        }
+    }
+
+    private class QueueFlush implements Runnable {
+
+        /** @see java.lang.Runnable#run() */
```

Comment:
Yes, it is an auto-generated comment, but IMO it makes sense it is specific case. It just redirects to the interface method JavaDoc, because we simply fulfill that contract and there is nothing more to say here.